subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c

```diff
@@ -120,50 +112,36 @@ fs_mgmt_file_download(struct mgmt_ctxt *ctxt)
 static int
 fs_mgmt_file_upload(struct mgmt_ctxt *ctxt)
 {
-	uint8_t file_data[CONFIG_FS_MGMT_UL_CHUNK_SIZE];
 	char file_name[CONFIG_FS_MGMT_PATH_SIZE + 1];
-	unsigned long long len;
-	unsigned long long off;
-	size_t data_len;
+	unsigned long long len = ULLONG_MAX;
+	unsigned long long off = ULLONG_MAX;
 	size_t new_off;
+	bool ok;
 	int rc;
-
-	const struct cbor_attr_t uload_attr[5] = {
-		[0] = {
-			.attribute = "off",
-			.type = CborAttrUnsignedIntegerType,
-			.addr.uinteger = &off,
-			.nodefault = true
-		},
-		[1] = {
-			.attribute = "data",
-			.type = CborAttrByteStringType,
-			.addr.bytestring.data = file_data,
-			.addr.bytestring.len = &data_len,
-			.len = sizeof(file_data)
-		},
-		[2] = {
-			.attribute = "len",
-			.type = CborAttrUnsignedIntegerType,
-			.addr.uinteger = &len,
-			.nodefault = true
-		},
-		[3] = {
-			.attribute = "name",
-			.type = CborAttrTextStringType,
-			.addr.string = file_name,
-			.len = sizeof(file_name)
-		},
-		[4] = { 0 },
+	zcbor_state_t *zse = ctxt->cnbe->zs;
+	zcbor_state_t *zsd = ctxt->cnbd->zs;
+	struct zcbor_string name = { 0 };
+	struct zcbor_string file_data = { 0 };
+	size_t decoded = 0;
+
+	struct zcbor_map_decode_key_val fs_upload_decode[] = {
+		ZCBOR_MAP_DECODE_KEY_VAL(off, zcbor_uint64_decode, &off),
+		ZCBOR_MAP_DECODE_KEY_VAL(name, zcbor_tstr_decode, &name),
+		ZCBOR_MAP_DECODE_KEY_VAL(data, zcbor_bstr_decode, &file_data),
+		ZCBOR_MAP_DECODE_KEY_VAL(len, zcbor_uint64_decode, &len),
 	};
 
-	len = ULLONG_MAX;
-	off = ULLONG_MAX;
-	rc = cbor_read_object(&ctxt->it, uload_attr);
-	if (rc != 0 || off == ULLONG_MAX || file_name[0] == '\0') {
+	ok = zcbor_map_decode_bulk(zsd, fs_upload_decode,
+		ARRAY_SIZE(fs_upload_decode), &decoded) == 0;
```

Comment:
`ok` is a bool and `zcbor_map_decode_bulk` is returning an int, is a case needed?

Reply:
no ==  does the cast

Reply:
Missed that!